src/PrestaShopBundle/Form/Partial/PartialForm.php

```diff
@@ -0,0 +1,319 @@
+ * that is bundled with this package in the file LICENSE.md.
+ * It is also available through the world-wide-web at this URL:
+ * https://opensource.org/licenses/OSL-3.0
+ * If you did not receive a copy of the license and are unable to
+ * obtain it through the world-wide-web, please send an email
+ * to license@prestashop.com so we can send you a copy immediately.
+ *
+ * DISCLAIMER
+ *
+ * Do not edit or add to this file if you wish to upgrade PrestaShop to newer
+ * versions in the future. If you wish to customize PrestaShop for your
+ * needs please refer to https://devdocs.prestashop.com/ for more information.
+ *
+ * @author    PrestaShop SA and Contributors <contact@prestashop.com>
+ * @copyright Since 2007 PrestaShop SA and Contributors
+ * @license   https://opensource.org/licenses/OSL-3.0 Open Software License (OSL 3.0)
+ */
+
+declare(strict_types=1);
+
+namespace PrestaShopBundle\Form\Partial;
+
+use Symfony\Component\EventDispatcher\EventDispatcherInterface;
+use Symfony\Component\Form\Exception\AlreadySubmittedException;
+use Symfony\Component\Form\Form;
+use Symfony\Component\Form\FormError;
+use Symfony\Component\Form\FormEvent;
+use Symfony\Component\Form\FormEvents;
+use Symfony\Component\Form\FormInterface;
+use Symfony\Component\Form\FormView;
+
+/**
+ * PartialForm is used to allow getting the data submitted to the Form.
+ *
+ * Since the parent class Form is hardly extendable because it has a lot of private
+ * fields this class is instead a wrapper. The main interest of this class is the
+ * getSubmittedData getter.
+ */
+class PartialForm implements \IteratorAggregate, PartialFormInterface
+{
+    /**
+     * @var Form
+     */
+    private $innerForm;
+
+    /**
+     * @var string|array|null
+     */
+    private $submittedData;
+
+    /**
+     * @param Form $form
+     * @param EventDispatcherInterface|null $dispatcher
+     */
+    public function __construct(Form $form, EventDispatcherInterface $dispatcher = null)
+    {
+        $this->innerForm = $form;
+
+        if (null !== $dispatcher && $dispatcher->hasListeners(FormEvents::PRE_SUBMIT)) {
+            // We set the listener internally so that the callback remains private
+            $dispatcher->addListener(
+                FormEvents::PRE_SUBMIT,
+                function (FormEvent $event) {
+                    $this->onSubmittedData($event);
+                },
+                // We set lowest priority so that it's called last and has latest modifications
+                -1000
+            );
+        }
+    }
+
+    /**
+     * @return Form
+     */
+    public function getInnerForm(): Form
+    {
+        return $this->innerForm;
+    }
+
+    /**
+     * @return array|string|null
+     */
+    public function getSubmittedData()
+    {
+        return $this->submittedData;
+    }
+
+    /**
+     * @return bool
+     */
+    public function usePartialUpdate(): bool
+    {
+        if (!$this->getConfig()->hasOption('use_partial_update')) {
```

Comment:
```suggestion
        if (!$this->getConfig()->hasOption(static::USE_PARTIAL_UPDATE)) {
```
Maybe use a const for this option name?

Reply:
It can't hurt indeed ^^